### DISCUSSION TITLE
Loading and pushing multiple models to GPUs at the same

### DISCUSSION THREAD

**Discussion by rahulvigneswaran** [Lightning-AI/pytorch-lightning#13023] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I have multiple pre-trained models at hand (one for each class). I have already loaded and frozen them. 

Now I need to extract features from them and distill them into a new model.

For doing so, am adding all the models to a python dict so that I can use the class label to access the appropriate model and extract features from them. But the problem is that these pre-trained models inside the python dict are not moved automatically to GPUs. What am I doing wrong here?

**Comment by lecacosa** [comment]
Maybe we could add models to a [`nn.ModuleDict`](https://pytorch.org/docs/stable/generated/torch.nn.ModuleDict.html#moduledict) instead of a python `dict`.
